core/src/main/java/com/linecorp/armeria/client/encoding/HttpDecodedResponse.java

```diff
@@ -45,12 +45,23 @@
     @Nullable
     private StreamDecoder responseDecoder;
     private boolean headersReceived;
+    private boolean decoderClosed;
 
     HttpDecodedResponse(HttpResponse delegate, Map<String, StreamDecoderFactory> availableDecoders,
                         ByteBufAllocator alloc) {
         super(delegate, true);
         this.availableDecoders = availableDecoders;
         this.alloc = alloc;
+        whenComplete().handle((unused, cause) -> {
+            if (decoderClosed) {
```

Comment:
I think this method could be called earlier than `beforeComplete()`. In that case, the `lastData` could not be produced?

Reply:
I realized that we always call `Subscriber.onComplete()` before calling `StreamMessage.completionFuture.complete(null)`. 😅